util/chplenv/printchplenv.py

```diff
@@ -91,7 +91,8 @@
     ChapelEnv('  CHPL_GPU_CODEGEN', RUNTIME | NOPATH),
     ChapelEnv('  CHPL_GPU_RUNTIME', RUNTIME | NOPATH, 'gpu'),
     ChapelEnv('  CHPL_CUDA_PATH', RUNTIME | NOPATH),
-    ChapelEnv('  CHPL_CUDA_LIBDEVICE_PATH', INTERNAL | NOPATH),
+    ChapelEnv('  CHPL_CUDA_LIBDEVICE_PATH', RUNTIME | NOPATH),
+    ChapelEnv('  CHPL_CUDA_MEMTYPE', RUNTIME | INTERNAL | NOPATH),
```

Comment:
I think I'd prefer to change this to `CHPL_GPU_MEM_STRATEGY`.

I think we'd want to support the same strategies for AMD so I wouldn't limit to calling this `CUDA_` something and this is really about more than just a single memory type.

It's really picking what memory we use for both memory allocation and other allocations (when on the GPU sublocale).  I think the term "strategy" communicates that there's some nuance to how this is done.

As far as valid values go I think there's a bigger debate there.
I want to try and avoid using the term "uva" because its confusing and not really telling the whole story.

I think I'll go with: `page_migration` and `array_data_on_device`.
Neither term is particularly good (`array_data_on_device` isn't telling the other half of the story which is non-array data is allocated in page-locked memory on the host).

Nevertheless I think it's a step in the right direction.